client/product-details/index.js

```diff
@@ -1,6 +1,42 @@
+/* global jQuery */
+// global wcpayStripeSiteMessaging
+
 /**
  * Internal dependencies
  */
-import bnplSiteMessaging from './bnpl-site-messaging';
+import { initializeBnplSiteMessaging } from './bnpl-site-messaging';
+
+const bnplPaymentMessageElement = initializeBnplSiteMessaging();
+
+const { productPrices } = window.wcpayStripeSiteMessaging;
+let { productId } = window.wcpayStripeSiteMessaging;
+
+jQuery( function ( $ ) {
+	$( '.quantity input' ).on( 'change', function ( event ) {
+		const newQuantity = event.target.value;
+		const price = productPrices[ productId ];
+		bnplPaymentMessageElement.update( {
+			amount: parseInt( price, 10 ) * newQuantity,
+		} );
+	} );
+
+	$( '.single_variation_wrap' ).on( 'show_variation', function (
```

Comment:
Would it make sense to wrap the variations bindings into `if ( productPrices.length > 1)`?

@KarlisJ, WDYT?

Reply:
That should save some CPU cycles for non-variation products.

Reply:
Done here! - 97f1a29

Reply:
Thanks! Should we move `$( '.reset_variations' ).on( 'click', function () {` also inside the if?

Reply:
Sorry about missing this. Indeed the condition applies only to variable products. I have moved it within the `if` statement. 
ef11e5a